src/createDataDictionary.m

```diff
@@ -13,30 +13,14 @@ function createDataDictionary(cfg, logFile)
     jsonContent = setJsonContent(fullFilename, logFile);
 
     opts.Indent = '    ';
+    
     bids.util.jsonencode(fullFilename, jsonContent, opts);
 
 end
 
 function jsonContent = setJsonContent(fullFilename, logFile)
```

Comment:
In fact, I think the field logFile.isStim should be definitely set here, because we only call createDataDictionary after saveEventsFile('init') has been performed. 

So is it necessary to check here whether isStim is a field? Maybe we can directly use logFile.isStim in the if statement, instead of creating another local variable? 

`    % add default _event file fields to the json content
    jsonContent = logFile.columns; 

    % overwrite json fields if this is a _stim file
    if logFile.isStim
        % ...
    end
        
    % transfer content of extra fields to json content
    namesExtraColumns = returnNamesExtraColumns(logFile);
    for iExtraColumn = 1:numel(namesExtraColumns)
        nbCol = returnNbColumns(logFile, namesExtraColumns{iExtraColumn});
        for iCol = 1:nbCol
            headerName = returnHeaderName(namesExtraColumns{iExtraColumn}, nbCol, iCol);
            if logFile.isStim
                jsonContent.Columns{end + 1} = headerName;
            end
            jsonContent.(headerName) = ...
                logFile(1).extraColumns.(namesExtraColumns{iExtraColumn}).bids;
        end
    end
`

Reply:
Whoops, another thing we can't do is call `logFile.columns` when we're making a _stim file, because this is not defined. In fact, we **only** have `logFile.extraColumns` in case of _stim file, so this would crash. 

Two solutions: 
1) simple 
```
    % overwrite json fields if this is a _stim file
    if logFile.isStim
        % ... set stuff for _stim file
    else
        jsonContent = logFile.columns; 
    end
```
2) more invasive
We could merge `logFile.columns` and `logFile.extraColumns` into a single subfield, i.e. fundamentally change the way we define columns during initialization and opening (but this would in turn require changing many other functions in the library)

Reply:
> In fact, I think the field logFile.isStim could be directly used here, because we only call createDataDictionary after saveEventsFile('init') has been performed.
So is it necessary to check here whether isStim is a field? Maybe we can directly use logFile.isStim in the if statement, instead of creating another local variable?

True we don't necessarily need to create a new variable but we should at least add a check to make sure that `isStim` is a non empty field of `logFile`: it is true that in this use case we call `createDataDictionary` after `saveEventsFile` but it might not always be the case so we want a way to define a default behavior for this function when `logFile.isStim` is not set.

Reply:
> Whoops, another thing we can't do is call `logFile.columns` when we're making a _stim file, because this is not defined. In fact, we **only** have `logFile.extraColumns` in case of _stim file, so this would crash.

Wait... I don't think we call `logFile.columns` when we we are in the `stim` case.

Reply:
> True we don't necessarily need to create a new variable but we should at least add a check to make sure that isStim is a non empty field of logFile: it is true that in this use case we call createDataDictionary after saveEventsFile but it might not always be the case so we want a way to define a default behavior for this function when logFile.isStim is not set.

Makes sense. I've updated the if-statements accordingly.

Reply:
> Wait... I don't think we call logFile.columns when we we are in the stim case.
You're right, sorry, nevermind.